WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -128,6 +133,27 @@ class MainActivity : AppCompatActivity() {
         val jsObjName = "jsObject"
         val allowedOriginRules = setOf<String>("https://gcoleman799.github.io")
 
+        // FORCE DARK
+        // Check if the system is set to light or dark mode
+        val nightModeFlag = resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK
+        if (nightModeFlag == Configuration.UI_MODE_NIGHT_YES) {
+            // Switch WebView to dark mode; uses default dark theme
+            if (WebViewFeature.isFeatureSupported(WebViewFeature.FORCE_DARK)) {
+                WebSettingsCompat.setForceDark(
+                    binding.webview.settings,
+                    WebSettingsCompat.FORCE_DARK_ON
+                )
+            }
+            // Use custom Dark Theme
+            // DARK_STRATEGY_WEB_THEME_DARKENING_ONLY - it never applies automatic darkening and uses web page only darkening.
+            if (WebViewFeature.isFeatureSupported(WebViewFeature.FORCE_DARK_STRATEGY)) {
+                WebSettingsCompat.setForceDarkStrategy(
+                    binding.webview.settings,
+                    DARK_STRATEGY_PREFER_WEB_THEME_OVER_USER_AGENT_DARKENING
```

Comment:
So, yesterday I learned there's a rather subtle difference between two of the strategies:

1. `PREFER_WEB_THEME_OVER_USER_AGENT_DARKENING` checks for the `<meta>` tag. If present, it uses media queries. If absent, it applies user-agent (automatic) darkening.
2. `WEB_THEME_DARKENING_ONLY` uses media queries always, even if there's no `<meta>` tag.

I think the "even if there's no `<meta>` tag" part is very surprising, and it might be a major "gotcha" if the app dev tries to switch strategies from `WEB_THEME_DARKENING_ONLY` to `PREFER_WEB_THEME_OVER_USER_AGENT_DARKENING`. It would be good to call out this detail.

In general, I think this is a good opportunity to add some comments to provide guidance for choosing between these two similar strategies.

Reply:
I agree that this would be a good place to explain some of these differences. I have added a comment, let me know if you think it offers enough clarification